### PR TITLE
task17 | added solution

### PR DESCRIPTION
**Pull request by TatsiaA**
task17 | added solution

### CODE REVIEW

File: tests/Task17/Task17test.php [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        $this->assertEquals($output, get1fromN($num));
+    }
+
+    public function numbersProvider(): array
+    {
+        return [
+            'get 1 from 12' => [
+                12,
+                [
+                    '0' => 12,
```
Comment (Jampire) [inline tests/Task17/Task17test.php:43]: Array keys are numbers, not strings.

File: src/Task17/task17.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ * @throws ErrorException
+ */
+function get1fromN(int $n): array
+{
+    if ($n <= 0) {
+        throw new ErrorException(sprintf('You can only pass positive integer (not %s)', $n));
+    }
+    $arr = [$n];
+    while ($n > 1) {
+        $n = $n % 2
```
Comment (Jampire) [inline src/Task17/task17.php:21]: Always do strong comparison. So, here should be `$n % 2 === 0` or `$n %2 !== 0`.

**Review by Jampire — CHANGES_REQUESTED** [review]

File: tests/Task17/Task17test.php (hunk 1 above)
Comment (TatsiaA) [reply tests/Task17/Task17test.php]: changed

File: src/Task17/task17.php (hunk 2 above)
Comment (TatsiaA) [reply src/Task17/task17.php]: corrected

**Pull request merged**